onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/webgpu/quantization/matmul_nbits.h"
+#include "contrib_ops/webgpu/webgpu_contrib_kernels.h"
+#include "core/providers/cpu/math/matmul_helper.h"
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace webgpu {
+
+namespace {
+// Put it to a common place?
```

Comment:
> Put it to a common place?

Maybe not.

My understanding is, this function is doing quite simple thing. It maybe can be used in multiple places, but the problem is, a developer may not try to find an existing solution if they have the exact same requirement, because writing one is perhaps easier than find the existing one.

This may add a quite maintenance cost because it requires developers and code reviewers to be very familiar with the code base so that they can figure out there is a `GetMaxComponents` function there.

I would say that a shared helper function need to be not too small/simple to be considered worth to put somewhere for sharing, and this function is not reaching that complexity.

Reply:
Update: had a discussion in today’s dev meeting about this. It is mentioned that this function together with a few others may be used for a quite different places. Decision is to be made later.